xwiki-platform-core/xwiki-platform-annotation/xwiki-platform-annotation-ui/src/main/resources/AnnotationCode/Script.xml

```diff
@@ -976,28 +977,193 @@ XWiki.Annotation = Class.create({
+   * with these new nodes.
+   *
+   * @param range the DOM range of the annotated text
+  */
+  fixRangeEnds: function(range) {
+    var strictRange = new Range();
+
+    var newStartNode = range.startContainer.splitText(range.startOffset);
+    strictRange.setStart(newStartNode, 0);
+
+    var newEndNode = range.endContainer.splitText(range.endOffset);
+    strictRange.setEnd(newEndNode, 0);
+
+    return strictRange;
+  },
+
+  /**
+   * Create a DOM Range by knowing the start and end index from inside the plain content of the element.
+   *
+   * @param annotatedElement the element from where the range is constructed
+   * @param startIndex start offset where the range begins
+   * @param endIndex end offset where the range end
+  */
+  getDOMRange: function(annotatedElement, startIndex, endIndex) {
+    this.annotatedElementPlainText = "";
+    var startPosition = this.getElementAtOffset(annotatedElement, startIndex);
+    this.annotatedElementPlainText = "";
+    var endPosition = this.getElementAtOffset(annotatedElement, endIndex);
+    var range = new Range();
+    range.setStart(startPosition.element, startPosition.offset);
+    range.setEnd(endPosition.element, endPosition.offset);
+    return range;
+  },
+
+  /**
+   * Knowing the offset relative to the full plain content of the root element, get the corresponding child node that contains it and the offset
+   * specific to the new node. The DOM is traversed recursively starting with the root node and the plain content is computed to know when the
+   * wanted element is found.
+   *
+   * @param rootElement the element where the search is done
+   * @param offset the offset relative to the full plain content
+  */
+  getElementAtOffset: function(rootElement, offset) {
+    var childNodes = rootElement.childNodes;
+    var child;
+    for (let i = 0; i &lt; childNodes.length; i++) {
+      child = childNodes[i];
+      if (child &amp;&amp; child.nodeType == 3) {
+        var initialLength = this.annotatedElementPlainText.length;
+        this.annotatedElementPlainText += child.textContent.replace(/\s/g, '');
+        if (offset &lt; this.annotatedElementPlainText.length) {
+          var elementSpecificOffset = this.getElementSpecificOffset(child, offset - initialLength);
+          return {"element": child, 'offset': elementSpecificOffset};
+        }
+      }
+      if (child.childNodes.length &gt; 0) {
+        var maybeFoundElement = this.getElementAtOffset(child, offset);
+        if (maybeFoundElement) {
+          return maybeFoundElement;
+        }
+      }
+    }
+  },
+
+  /**
+   * Get a offset knowing the offset relatively to the plain text of the element.
+   *
+   * @param node a DOM node
+   * @param offset the offset relative to the plain text
+  */
+  getElementSpecificOffset: function(node, offset) {
+    var nonSpaceCharsLength = 0;
+    var chars = Array.from(node.textContent);
+    for (let i = 0; i &lt; chars.length; i++) {
+      if (!/\s/.test(chars[i])) {
```

Comment:
I don't understand why you need to ignore spaces.

Reply:
Since the given offset is relative to the plain content of the text node. What I am trying is to find where that offset is considering also the spaces.